### PR TITLE
util.cc: refactor read_file

### PR DESCRIPTION
**Pull request by deanlee**
1. Open with read mode flag
2. Open the file only once

### CODE REVIEW

File: selfdrive/common/util.cc [hunk 1]
```diff
@@ -54,26 +54,20 @@ int set_core_affinity(int core) {
 namespace util {
 
 std::string read_file(const std::string& fn) {
-  std::ifstream ifs(fn, std::ios::binary | std::ios::ate);
-  if (ifs) {
-    int pos = ifs.tellg();
-    if (pos > 0) {
+  if (std::ifstream f{fn, std::ios::binary | std::ios::in | std::ios::ate}) {
+    if (int pos = f.tellg(); pos > 0) {
```
Comment (adeebshihadeh) [inline selfdrive/common/util.cc:58]:
```suggestion
    int pos = f.tellg();
    if (ipos > 0) {
```

File: selfdrive/common/util.cc [hunk 2]
```diff
@@ -54,26 +54,20 @@ int set_core_affinity(int core) {
 namespace util {
 
 std::string read_file(const std::string& fn) {
-  std::ifstream ifs(fn, std::ios::binary | std::ios::ate);
-  if (ifs) {
-    int pos = ifs.tellg();
-    if (pos > 0) {
+  if (std::ifstream f{fn, std::ios::binary | std::ios::in | std::ios::ate}) {
```
Comment (adeebshihadeh) [inline selfdrive/common/util.cc:57]:
```suggestion
  std::ifstream f{fn, std::ios::binary | std::ios::in | std::ios::ate}
  if (f) {
```

File: selfdrive/common/util.cc [hunk 3]
```diff
@@ -54,26 +54,22 @@ int set_core_affinity(int core) {
...
     if (pos > 0) {
       std::string result;
       result.resize(pos);
-      ifs.seekg(0, std::ios::beg);
-      ifs.read(result.data(), pos);
-      if (ifs) {
-        return result;
-      }
+      f.seekg(0, std::ios::beg);
+      return f.read(result.data(), pos) ? result : std::string();
```
Comment (adeebshihadeh) [inline selfdrive/common/util.cc:64]:
```suggestion
      if (f) {
        return result;
      }
```

File: selfdrive/common/util.cc [hunk 4]
```diff
@@ -61,7 +61,9 @@ std::string read_file(const std::string& fn) {
       std::string result;
       result.resize(pos);
       f.seekg(0, std::ios::beg);
-      return f.read(result.data(), pos) ? result : std::string();
+      if (f) {
```
Comment (deanlee) [inline selfdrive/common/util.cc:64]: if (f.read(result.data(), pos)) {

**Comment by adeebshihadeh** [comment]
Thanks for writing the test first :)

**Pull request merged**